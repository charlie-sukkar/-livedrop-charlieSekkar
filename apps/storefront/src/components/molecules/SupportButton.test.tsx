import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { describe, it, expect, vi } from 'vitest';
import { SupportButton } from './SupportButton';

// Create the mock function first
const mockOpenSupport = vi.fn();

// Mock the store - return the function directly
vi.mock('../../lib/store', () => ({
  useSupportStore: () => mockOpenSupport, // Return function directly, not object
}));

// Mock Button component - ensure onClick is called properly
vi.mock('../atoms/Button', () => ({
  Button: ({ children, onClick, disabled, className, 'aria-label': ariaLabel }: any) => (
    <button 
      onClick={() => onClick?.()} // Ensure onClick is called as function
      disabled={disabled}
      className={className}
      aria-label={ariaLabel}
    >
      {children}
    </button>
  ),
}));

// Mock Icon component  
vi.mock('../atoms/Icon', () => ({
  Icon: ({ className, d }: any) => (
    <svg className={className} data-testid="support-icon" data-d={d} />
  ),
}));

describe('SupportButton', () => {
  beforeEach(() => {
    mockOpenSupport.mockClear();
  });

  // Props Testing
  it('renders with default props', () => {
    render(<SupportButton />);
    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  it('applies custom className', () => {
    render(<SupportButton className="custom-class" />);
    const button = screen.getByRole('button');
    expect(button).toHaveClass('custom-class');
    expect(button).toHaveClass('relative');
  });

  it('respects disabled prop', () => {
    render(<SupportButton disabled={true} />);
    expect(screen.getByRole('button')).toBeDisabled();
  });

  // User Events Testing
  it('calls openSupport when clicked', async () => {
    const user = userEvent.setup();
    render(<SupportButton />);

    await user.click(screen.getByRole('button'));
    expect(mockOpenSupport).toHaveBeenCalledOnce();
  });

  it('does not call openSupport when disabled', async () => {
    const user = userEvent.setup();
    render(<SupportButton disabled={true} />);

    await user.click(screen.getByRole('button'));
    expect(mockOpenSupport).not.toHaveBeenCalled();
  });

  // Accessibility Testing
  it('has proper ARIA label', () => {
    render(<SupportButton />);
    expect(screen.getByLabelText('Open support assistant')).toBeInTheDocument();
  });

  // Component Integration
  it('renders icon with correct props', () => {
    render(<SupportButton />);
    const icon = screen.getByTestId('support-icon');
    expect(icon).toHaveClass('h-6 w-6');
    expect(icon).toHaveAttribute('data-d');
  });
});