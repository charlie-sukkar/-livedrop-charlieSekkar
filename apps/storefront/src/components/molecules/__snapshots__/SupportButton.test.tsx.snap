// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SupportButton > matches custom className snapshot 1`] = `
<button
  aria-label="Open support assistant"
  class="relative border-2"
  data-testid="support-button"
>
  <svg
    class="h-6 w-6"
    data-d="M8.228 9c.549-1.165 2.03-2 3.772-2 2.21 0 4 1.343 4 3 0 1.4-1.278 2.575-3.006 2.907-.542.104-.994.54-.994 1.093m0 3h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
    data-testid="support-icon"
  />
</button>
`;

exports[`SupportButton > matches default state snapshot 1`] = `
<button
  aria-label="Open support assistant"
  class="relative "
  data-testid="support-button"
>
  <svg
    class="h-6 w-6"
    data-d="M8.228 9c.549-1.165 2.03-2 3.772-2 2.21 0 4 1.343 4 3 0 1.4-1.278 2.575-3.006 2.907-.542.104-.994.54-.994 1.093m0 3h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
    data-testid="support-icon"
  />
</button>
`;

exports[`SupportButton > matches disabled state snapshot 1`] = `
<button
  aria-label="Open support assistant"
  class="relative "
  data-testid="support-button"
  disabled=""
>
  <svg
    class="h-6 w-6"
    data-d="M8.228 9c.549-1.165 2.03-2 3.772-2 2.21 0 4 1.343 4 3 0 1.4-1.278 2.575-3.006 2.907-.542.104-.994.54-.994 1.093m0 3h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
    data-testid="support-icon"
  />
</button>
`;
